src/webhooks/registry.ts

```diff
@@ -107,11 +142,14 @@ function buildQuery(
   let mutationName: string;
   let webhookSubscriptionArgs: string;
   switch (deliveryMethod) {
-    case DeliveryMethod.Http:
+    case DeliveryMethod.Http || !versionSupportsEndpointField():
```

Comment:
If the version is pre-`2020-07` and `deliveryMethod` is passed in with `DeliveryMethod.EventBridge`, the result will be a GraphQL query built for `Http`, i.e., with the mutation being `webhookSubscriptionUpdate` or `webhookSubscriptionCreate` and `callbackUrl` being the `arn` passed in, though it should be (for a Http type query), a portion of a URL that is internally prefixed with `https://${Context.HOST_NAME}`.

This will result in a malformed query.

I think the check
```
if (deliveryMethod === DeliveryMethod.EventBridge && !versionSupportsEndpointField()) {
  // throw an error
}
```
should be added just after the method invocation (i.e., throw out the mismatched scenario early)

Reply:
Oh yes, good catch. I've added that in 22e510f25622dd6cf8c5fda05690b89efee79383.